corelib-definitions/src/main/java/eu/europeana/corelib/definitions/model/facets/logic/ImageTagExtractor.java

```diff
@@ -0,0 +1,261 @@
+        hexColors.put("#32CD32",77);
+        hexColors.put("#FAF0E6",78);
+        hexColors.put("#800000",79);
+        hexColors.put("#66CDAA",80);
+        hexColors.put("#0000CD",81);
+        hexColors.put("#BA55D3",82);
+        hexColors.put("#9370DB",83);
+        hexColors.put("#3CB371",84);
+        hexColors.put("#7B68EE",85);
+        hexColors.put("#00FA9A",86);
+        hexColors.put("#48D1CC",87);
+        hexColors.put("#C71585",88);
+        hexColors.put("#191970",89);
+        hexColors.put("#F5FFFA",90);
+        hexColors.put("#FFE4E1",91);
+        hexColors.put("#FFE4B5",92);
+        hexColors.put("#FFDEAD",93);
+        hexColors.put("#000080",94);
+        hexColors.put("#FDF5E6",95);
+        hexColors.put("#808000",96);
+        hexColors.put("#6B8E23",97);
+        hexColors.put("#FFA500",98);
+        hexColors.put("#FF4500",99);
+        hexColors.put("#DA70D6",100);
+        hexColors.put("#EEE8AA",101);
+        hexColors.put("#98FB98",102);
+        hexColors.put("#AFEEEE",103);
+        hexColors.put("#DB7093",104);
+        hexColors.put("#FFEFD5",105);
+        hexColors.put("#FFDAB9",106);
+        hexColors.put("#CD853F",107);
+        hexColors.put("#FFC0CB",108);
+        hexColors.put("#DDA0DD",109);
+        hexColors.put("#B0E0E6",110);
+        hexColors.put("#800080",111);
+        hexColors.put("#FF0000",112);
+        hexColors.put("#BC8F8F",113);
+        hexColors.put("#4169E1",114);
+        hexColors.put("#8B4513",115);
+        hexColors.put("#FA8072",116);
+        hexColors.put("#F4A460",117);
+        hexColors.put("#2E8B57",118);
+        hexColors.put("#FFF5EE",119);
+        hexColors.put("#A0522D",120);
+        hexColors.put("#C0C0C0",121);
+        hexColors.put("#87CEEB",122);
+        hexColors.put("#6A5ACD",123);
+        hexColors.put("#708090",124);
+        hexColors.put("#FFFAFA",125);
+        hexColors.put("#00FF7F",126);
+        hexColors.put("#4682B4",127);
+        hexColors.put("#D2B48C",128);
+        hexColors.put("#008080",129);
+        hexColors.put("#D8BFD8",130);
+        hexColors.put("#FF6347",131);
+        hexColors.put("#40E0D0",132);
+        hexColors.put("#EE82EE",133);
+        hexColors.put("#F5DEB3",134);
+        hexColors.put("#FFFFFF",135);
+        hexColors.put("#F5F5F5",136);
+        hexColors.put("#FFFF00",137);
+        hexColors.put("#9ACD32",138);
+    }
+
+    public static Integer getSizeCode(final Integer width, final Integer height) {
+        if(width == null || height == null) {
+            return 0;
+        }
+
+        final Long size = (long) (width * height);
+        if(size < 524288) {
+            return 1;
+        }
+        if(size < 1048576) {
+            return 2;
+        }
+        if(size < 4194304) {
+            return 3;
+        }
+
+        return 4;
+    }
+
+    public static Integer getSizeCode(final String imageSize) {
+        if(imageSize == null || imageSize.equals("")) {
```

Comment:
By reversing the equals below if ("small".equals(imageSize)) ..., this statement is not needed...

But better is to use the java 7 string switch instead and use the default for return 0
switch (StringUtils.lowercase(imageSize)) {
case "small": 
  return 1;
case "medium": 
  return 2;
....
default: 
  return 0;